data_pipelines/rne/database/process_rne.py

```diff
@@ -0,0 +1,294 @@
+):
+    connection, cursor = connect_to_db(db_path)
+
+    for dirigeant_pp in list_dirigeants_pp:
+        find_and_delete_same_siren(cursor, dirigeant_pp["siren"], file_path)
+        cursor.execute(
+            """
+            INSERT INTO dirigeants_pp (siren,
+            date_mise_a_jour, date_de_naissance,
+            role, nom,
+                                        nom_usage,
+                                        prenoms, genre, nationalite,
+                                        situation_matrimoniale,
+                                        pays, code_pays, code_postal,
+                                        commune, code_insee_commune, voie, file_name)
+            VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
+        """,
+            (
+                dirigeant_pp["siren"],
+                dirigeant_pp["date_mise_a_jour"],
+                dirigeant_pp["date_de_naissance"],
+                dirigeant_pp["role"],
+                dirigeant_pp["nom"],
+                dirigeant_pp["nom_usage"],
+                dirigeant_pp["prenoms"],
+                dirigeant_pp["genre"],
+                dirigeant_pp["nationalite"],
+                dirigeant_pp["situation_matrimoniale"],
+                dirigeant_pp["pays"],
+                dirigeant_pp["code_pays"],
+                dirigeant_pp["code_postal"],
+                dirigeant_pp["commune"],
+                dirigeant_pp["code_insee_commune"],
+                dirigeant_pp["voie"],
+                file_path,
+            ),
+        )
+
+    for dirigeant_pm in list_dirigeants_pm:
+        find_and_delete_same_siren(cursor, dirigeant_pm["siren"], file_path)
+        cursor.execute(
+            """
+            INSERT INTO dirigeants_pm
+            (siren, date_mise_a_jour, denomination, siren_dirigeant, role,
+                                        forme_juridique, pays, code_pays, code_postal,
+                                        commune, code_insee_commune, voie, file_name)
+            VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
+        """,
+            (
+                dirigeant_pm["siren"],
+                dirigeant_pm["date_mise_a_jour"],
+                dirigeant_pm["denomination"],
+                dirigeant_pm["siren_dirigeant"],
+                dirigeant_pm["role"],
+                dirigeant_pm["forme_juridique"],
+                dirigeant_pm["pays"],
+                dirigeant_pm["code_pays"],
+                dirigeant_pm["code_postal"],
+                dirigeant_pm["commune"],
+                dirigeant_pm["code_insee_commune"],
+                dirigeant_pm["voie"],
+                file_path,
+            ),
+        )
+    cursor.execute("SELECT COUNT(*) FROM dirigeants_pp")
+    count_pp = cursor.fetchone()[0]
+
+    cursor.execute("SELECT COUNT(*) FROM dirigeants_pm")
+    count_pm = cursor.fetchone()[0]
+
+    logging.info(f"************Count pp: {count_pp}, Count pm: {count_pm}")
```

Comment:
We can add a sanity check : you can compare `count_XX` to `len(list_dirigeants_XX)` and raise an exception if it doesnot match.

Reply:
The `list_dirigeants_XX` changes for every iteration and is not cumulative, so that check will not be accurate.
I want however to add a final check to see if database has more than, say 10million records. What do you think ?

Reply:
I would rather raise an exception fi it has less than a threshold
But yes seems a good idea

Reply:
I added a task that checks for the count before uploading the database to MinIO.